src/data/nat/prime.lean

```diff
@@ -645,6 +645,16 @@ have hn : 0 < n := nat.pos_of_ne_zero $ λ h, begin
 end,
 perm_of_prod_eq_prod (by rwa prod_factors hn) h₂ (@prime_of_mem_factors _)
 
+lemma prime.factors_pow {p : ℕ} (hp : p.prime) (y : ℕ) :
+  (p ^ y).factors = list.repeat p y :=
+begin
+  symmetry,
+  rw ← list.repeat_perm,
+  apply nat.factors_unique (list.prod_repeat p y),
+  { intros q hq,
+    rwa eq_of_mem_repeat hq },
+end
+
```

Comment:
Can you add another lemma that describes the trivial consequence of this?
```
lemma prime.factors_eq {p : ℕ} (hp : p.prime) :
  p.factors = [p] :=
```

Reply:
Oh never mind, we already have that, it's `factors_prime`.

Reply:
This already exists on line 412 and is currently named factors_prime